### PR TITLE
Optimize id and class with escaping characters

### PR DESCRIPTION
**Pull request by NN---**
Fixes https://github.com/jquery/jquery/issues/3937

### CODE REVIEW

**Comment by NN---** [comment]
Seems like a problem with JSHint: https://github.com/jshint/jshint/issues/3238

File: src/sizzle.js [hunk 1]
```diff
@@ -128,7 +128,18 @@ var i,
 	rnative = /^[^{]+\{\s*\[native \w/,
 
 	// Easily-parseable/retrievable ID or TAG or CLASS selectors
-	rquickExpr = /^(?:#([\w-]+)|(\w+)|\.([\w-]+))$/,
+	// ID: #((?:[\w-]|\\[\!\"\#\$\%\&\'\(\)\*\+\,\-\.\/\:\;\<\=\>\?\@\\\[\]\^\`\{\|\}\~]|\\[a-f0-9]{1,6})+)
```
Comment (dmethvin) [inline src/sizzle.js]: A regexp this complex would need to be thoroughly tested in the unit tests, and there are no additional tests here. I also don't see any perf test.

**Comment by timmywil** [comment]
Thank you, but I don't think we should make this change.

File: src/sizzle.js (hunk 1 above)
Comment (NN---) [reply src/sizzle.js]: I can add it. 
I ran on my own website and it has factor of 1000 in IE.

**Comment by NN---** [comment]
@timmywill ok.
Perhaps do you know why jQuery calls uniqueSort after querying using getElementsByClassName ?
Do browsers really return duplicates in that function?
Thanks.

**Comment by NN---** [comment]
I made a much simpler regex which should cover most popular cases.
Please review.

**Pull request closed without merging**